### ISSUE TITLE
High CPU load on newish install (v1.23.5 / arm64 / Ubuntu 22.04)

### ISSUE THREAD

**Issue by mprossau** [canonical/microk8s#3087]
Hi,

This is my first attempt at building a microk8s install on standalone hardware. I am using three Raspberry PI 4B 8GB models connected via a Ubiquiti ethernet switch. Am observing high CPU utilization (most of the nodes are sitting at 100%+). Best guess is it's something to do with the networking.

Output of hostnamectl on the main node is below (other two nodes are identical with the exception of different hostnames). DNS is handled via a host file in each node. Am using microSD cards for storage on each of the devices. I followed the tutorial from here https://ubuntu.com/tutorials/how-to-kubernetes-cluster-on-raspberry-pi#4-installing-microk8s

--
 Static hostname: k8s-node-1
       Icon name: computer
      Machine ID: 6093d688d7da43a8a57c215d962a0d13
         Boot ID: 43e5b601c3a940ba83a4e758a12a4b61
Operating System: Ubuntu 22.04 LTS
          Kernel: Linux 5.15.0-1005-raspi
    Architecture: arm64
--

Things of interest I have found from investigating this:

1. There is a noticeable pause when executing microk8s commands.
2. Top three processes by CPU load are calico-node, k8s-dqlite and kubelite.
3. Am seeing multiple warnings (WARNING:Unknown index 67239 seen, reloading interface list) and errors (ERROR:Unknown interface index 67265 seen even after reload) in the logs from networkd-dispatcher. 

Copy of the inspection report is attached. Am assuming this sort of CPU load shouldn't be there on a newish install. Not certain where to do go next and given Ubuntu 22.04 is very new thought I'd reach out here. 

[inspection-report-20220423_094411.tar.gz](https://github.com/canonical/microk8s/files/8544824/inspection-report-20220423_094411.tar.gz)

**Comment by balchua** [comment]
From what i can see, 100% cpu seems to indicate that one or more of the components are crashlooping or aren't healthy.  Dqlite is a little sensitive to slow disk.  Some users have better success with non HA cluster.  If you don't need HA, then you can disable it. `microk8s disable ha-cluster`.
This will definitely nuke the contents.  So it is best to start from scratch if you want to give this a try.

Fyi, Kubernetes will continue to actively access its data store (etcd or dqlite) even if there is little workload running.

**Comment by mprossau** [comment]
I have investigated further and feel I have a better idea of what is occurring. Tentative root cause is later versions of Ubuntu for Raspberry Pi not including the vxlan kernel module by default. 

Adding the vxlan module required installing the following two ubuntu packages: _linux-modules-extra-5.15.0-1005-raspi linux-modules-extra-raspi_

Steps taken so far:

1. I deleted the existing cluster and rebuilt it with ha-cluster disabled. System load was better after that was done. 
2. Trying to re-enable dns, dashboard and ingress hit issues with a number of pods stuck at the creation stage.
3. Describing the failed pods gave the below error:
--
Warning  FailedCreatePodSandBox  5m1s (x106 over 28m)  kubelet  (combined from similar events): Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "0534e2955afd9c4bbcbcb58d15c7db215bc8643926a18ee7412e36908bb5609c": open /var/snap/microk8s/common/run/flannel/subnet.env: no such file or directory
--
4. Digging into the flannel logs gave the below error message. Main part of interest was '_VXLAN config: VNI=1 Port=0 GBP=false DirectRouting=false_' and '_Error registering network: operation not supported_'. This put me on the path of an issue with vxlan. 

--
Apr 24 16:25:11 k8s-node-1 microk8s.daemon-flanneld[379219]: PrevNode.Value: {"Network": "10.1.0.0/16", "Backend": {"Type": "vxlan"}}
Apr 24 16:25:11 k8s-node-1 microk8s.daemon-flanneld[379227]: {"Network": "10.1.0.0/16", "Backend": {"Type": "vxlan"}}
Apr 24 16:25:11 k8s-node-1 microk8s.daemon-flanneld[379072]: I0424 16:25:11.571747  379072 main.go:514] Determining IP address of default interface
Apr 24 16:25:11 k8s-node-1 microk8s.daemon-flanneld[379072]: I0424 16:25:11.572623  379072 main.go:527] Using interface with name eth0 and address 192.168.12.7
Apr 24 16:25:11 k8s-node-1 microk8s.daemon-flanneld[379072]: I0424 16:25:11.572691  379072 main.go:544] Defaulting external address to interface address (192.168.12.7)
Apr 24 16:25:11 k8s-node-1 flanneld[379072]: warning: ignoring ServerName for user-provided CA for backwards compatibility is deprecated
Apr 24 16:25:11 k8s-node-1 microk8s.daemon-flanneld[379072]: I0424 16:25:11.574521  379072 main.go:244] Created subnet manager: Etcd Local Manager with Previous Subnet: None
Apr 24 16:25:11 k8s-node-1 microk8s.daemon-flanneld[379072]: I0424 16:25:11.574558  379072 main.go:247] Installing signal handlers
Apr 24 16:25:11 k8s-node-1 microk8s.daemon-flanneld[379072]: I0424 16:25:11.645680  379072 main.go:386] Found network config - Backend type: vxlan
Apr 24 16:25:11 k8s-node-1 microk8s.daemon-flanneld[379072]: I0424 16:25:11.645835  379072 vxlan.go:120] VXLAN config: VNI=1 Port=0 GBP=false DirectRouting=false
Apr 24 16:25:11 k8s-node-1 microk8s.daemon-flanneld[379072]: E0424 16:25:11.652054  379072 main.go:289] Error registering network: operation not supported
Apr 24 16:25:11 k8s-node-1 microk8s.daemon-flanneld[379072]: I0424 16:25:11.652243  379072 main.go:366] Stopping shutdownHandler...
Apr 24 16:25:11 k8s-node-1 systemd[1]: snap.microk8s.daemon-flanneld.service: Main process exited, code=exited, status=1/FAILURE
Apr 24 16:25:11 k8s-node-1 systemd[1]: snap.microk8s.daemon-flanneld.service: Failed with result 'exit-code'.
Apr 24 16:25:11 k8s-node-1 systemd[1]: snap.microk8s.daemon-flanneld.service: Scheduled restart job, restart counter is at 4026.
Apr 24 16:25:11 k8s-node-1 systemd[1]: Stopped Service for snap application microk8s.daemon-flanneld.
--

5. Having a prod into the system made me realise the module wasn't available to install. Installing the extra files (above) cleared the issue and allowed the remaining pod creations to complete.
6. After this was confirmed as working I deleted the cluster and rebuilt it again with ha-cluster enabled. Have also enabled the dns, dashboard and ingress items. System is now sitting at ~35-40% CPU load.
